src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentExcerptService.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// excerpt some part of <see cref="Document"/>
+    /// </summary>
+    internal interface IDocumentExcerptService : IDocumentService
+    {
+        /// <summary>
+        /// return <see cref="ExcerptResult"/> of given <see cref="Document"/> and <see cref="TextSpan"/>
+        /// 
+        /// the result might not be an exact copy of the given source or contains more then given span
+        /// </summary>
+        Task<ExcerptResult?> TryExcerptAsync(Document document, TextSpan span, ExcerptMode mode, CancellationToken cancellationToken);
+    }
+
+    /// <summary>
+    /// this mode shows intention not actual behavior. it is up to implementation how to interpret the intention.
+    /// </summary>
+    internal enum ExcerptMode
+    {
+        SingleLine,
+        Tooltip
+    }
+
+    /// <summary>
+    /// Result of excerpt
+    /// </summary>
+    internal struct ExcerptResult
+    {
+        /// <summary>
+        /// excerpt content
+        /// </summary>
+        public readonly SourceText Content;
```

Comment:
i'm not sure what this means.  Is this just the text corresponding to the span?  to the mapped span?  to the original document?  do the lines excerpted depending on the excerpt mode?

Reply:
I am open to any better comments.

Reply:
Great.  For a start, the comment should address teh above questions i listed.  Thanks!